runtime/lua/man.lua

```diff
@@ -639,8 +639,20 @@ function M.init_pager()
   local _, sect, err = pcall(parse_ref, ref)
   vim.b.man_sect = err ~= nil and sect or ''
 
-  if not fn.bufname('%'):match('man://') then -- Avoid duplicate buffers, E95.
-    vim.cmd.file({ 'man://' .. fn.fnameescape(ref):lower(), mods = { silent = true } })
+  local man_bufname = 'man://' .. fn.fnameescape(ref):lower()
+
+  -- Raw manpage into (:Man!) overlooks `match('man://')` condition,
+  -- so if the buffer already exists, create new with a non existing name.
+  if vim.fn.bufexists(man_bufname) == 1 then
+    local new_bufname = man_bufname
+    while vim.fn.bufexists(new_bufname) == 1 do
+      new_bufname = new_bufname .. '_copy'
```

Comment:
the name is a uri. how about something like this,

```suggestion
      new_bufname = man_bufname .. '/' .. tostring(i)
```